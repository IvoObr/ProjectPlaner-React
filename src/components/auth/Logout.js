import React, {Component} from 'react';
import {actions} from "../../store/actions/authActions";
import {connect} from "react-redux";
import {Redirect} from "react-router-dom";

class Logout extends Component {
    componentDidMount() {
      this.props.logout();
    };

    render() {
            return (
                <Redirect to='/login'/>
            )
        }
}

const mapDispatchToProps = {
    logout: actions.logout
};

export default connect(null, mapDispatchToProps)(Logout);