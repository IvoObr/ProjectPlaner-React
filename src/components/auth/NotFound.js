import React from 'react';

export const NotFound = () => {
    return (
        <div className='container margin'>
            <div className='card padding-round'>
                <h3 className='card-title center'>Page not found</h3>
                <p className='center'><i className="material-icons large">stars</i></p>
            </div>
        </div>
    )
};


