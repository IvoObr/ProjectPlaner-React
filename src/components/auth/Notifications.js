import React from 'react';

const Notifications = (props) => {
    const doShow = props.doShow;
    const message = props.message;
    return (
        (doShow) ?
            <div className=' absolute hideElement center-align'>
                <div className='left-float '>
                    <i className="white-text material-icons line-height-left large">error_outline</i>
                </div>
                <div className='right-float'>
                    <p className='line-height-right'>{message}</p>
                </div>
            </div>
        : null
    )
};

export default Notifications;