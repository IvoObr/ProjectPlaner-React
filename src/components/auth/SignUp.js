import React, {Component} from 'react';
import {actions} from "../../store/actions/authActions";
import {connect} from "react-redux";
import {Redirect} from "react-router-dom";

class SignUp extends Component {
    state = {
        email: '',
        password: '',
        firstName: '',
        lastName: ''
    };

    handleChange = (event) => {
        this.setState({
            [event.target.id]: event.target.value
        })
    };

    handleSubmit = (event) => {
        event.preventDefault();
        this.props.signup(this.state);

    };

    render() {
        if (this.props.isLoggedIn) {
            return <Redirect to='/'/>
        }

        return (
            <div className='container'>
                <form onSubmit={this.handleSubmit} className='white'>
                    <h5 className="grey-text text-darken-3">Sign Up</h5>

                    <div className="input-field">
                        <label htmlFor="email">Email</label>
                        <input type="email" id='email' onChange={this.handleChange}/>
                    </div>

                    <div className="input-field">
                        <label htmlFor="password">Password</label>
                        <input type="password" id='password' onChange={this.handleChange}/>
                    </div>

                    <div className="input-field">
                        <label htmlFor="firstName">First Name</label>
                        <input type="text" id='firstName' onChange={this.handleChange}/>
                    </div>

                    <div className="input-field">
                        <label htmlFor="lastName">Last Name</label>
                        <input type="text" id='lastName' onChange={this.handleChange}/>
                    </div>


                    <div className="input-field">
                        <button className="btn green lighten-1 z-depth-0">Sign Up</button>
                    </div>
                    {this.props.loggingIn ? <div className="progress"> <div className="indeterminate"></div> </div>
                        : <div className="height-18"></div>}
                    {this.props.signUpError ?  <div className='hideElement absolute center-align'>
                        <i className="material-icons center-align large">error_outline</i>
                        <p >{this.props.signUpError}</p>
                    </div>
                        : null}
                </form>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        isLoggedIn: state.auth.loggedIn,
        loggingIn: state.auth.loggingIn,
        signUpError: state.auth.signUpError
    }
};

const mapDispatchToProps = {
    signup: actions.signup
};

export default connect(mapStateToProps, mapDispatchToProps)(SignUp);