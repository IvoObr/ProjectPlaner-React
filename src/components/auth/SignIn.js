import React, {Component} from 'react';
import {connect} from "react-redux";
import {actions} from "../../store/actions/authActions";
import {Redirect} from "react-router-dom";
import Notifications from "./Notifications";

class SignIn extends Component {
    state = {
        email: '',
        password: ''
    };

    handleChange = (event) => {
        this.setState({
            [event.target.id]: event.target.value
        })
    };

    handleSubmit = (event) => {
        event.preventDefault();
        this.props.login(this.state);
    };

    render() {
        if (this.props.isLoggedIn) {
           return <Redirect to='/'/>
        }

        return (
            <div className='container'>
                <form onSubmit={this.handleSubmit} className='white'>
                    <h5 className="grey-text text-darken-3">Login</h5>

                    <div className="input-field">
                        <label htmlFor="email">Email</label>
                        <input type="email" id='email' onChange={this.handleChange}/>
                    </div>

                    <div className="input-field">
                        <label htmlFor="password">Password</label>
                        <input type="password" id='password' onChange={this.handleChange}/>
                    </div>

                    <div className="input-field">
                        <button className="btn green lighten-1 z-depth-0">Login</button>
                    </div>
                        {this.props.loggingIn ? <div className="progress"> <div className="indeterminate"></div> </div> : <div className="height-19"></div>}
                        <Notifications doShow={this.props.loginError} message={'Invalid username or password'}/>
                    </form>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
      return {
          isLoggedIn: state.auth.loggedIn,
          loginError: state.auth.loginError,
          loggingIn: state.auth.loggingIn,
      }
};

const mapDispatchToProps = {
    login: actions.login
};

export default connect(mapStateToProps, mapDispatchToProps)(SignIn);