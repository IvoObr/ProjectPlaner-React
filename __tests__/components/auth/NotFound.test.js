import React from 'react';
import ReactDOM from 'react-dom';
import {NotFound} from '../../../src/components/auth/NotFound';
import {configure, shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({adapter: new Adapter()});

describe('<NotFound/>', () => {
    let wrapper = null;

    beforeEach(() => {
        wrapper = shallow(<NotFound/>);
    });

    it('should render NotFound', () => {
        expect(wrapper.find('div').length).toBe(2);
        expect(wrapper.find('h3').length).toBe(1);
    });

    it('matches snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
