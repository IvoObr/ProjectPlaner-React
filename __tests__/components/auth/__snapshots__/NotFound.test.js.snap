// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound/> matches snapshot 1`] = `
<div
  className="container margin"
>
  <div
    className="card padding-round"
  >
    <h3
      className="card-title center"
    >
      Page not found
    </h3>
    <p
      className="center"
    >
      <i
        className="material-icons large"
      >
        stars
      </i>
    </p>
  </div>
</div>
`;
